Auth0Tests/IDTokenSignatureValidatorSpec.swift

```diff
@@ -0,0 +1,112 @@
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import Foundation
+import Quick
+import Nimble
+import OHHTTPStubs
+
+@testable import Auth0
+
+class IDTokenSignatureValidatorSpec: IDTokenValidatorBaseSpec {
+    
+    override func spec() {
+        let domain = self.domain
+        
+        describe("signature validation") {
+            let signatureValidator = IDTokenSignatureValidator(context: validatorContext)
+            
+            context("algorithm support") {
+                beforeEach {
+                    stub(condition: isJWKSPath(domain)) { _ in jwksResponse() }
+                }
+                
+                it("should support RS256") {
+                    let jwt = generateJWT(alg: "RS256")
+                    
+                    waitUntil { done in
+                        signatureValidator.validate(jwt) { error in
+                            expect(error).to(beNil())
+                            done()
+                        }
+                    }
+                }
+
+                it("should not support other algorithms") {
+                    let alg = "AES256"
+                    let jwt = generateJWT(alg: alg)
+                    let expectedError = IDTokenSignatureValidator.ValidationError.invalidAlgorithm(actual: alg, expected: "RS256")
+                    
+                    waitUntil { done in
+                        signatureValidator.validate(jwt) { error in
+                            expect(error).to(matchError(expectedError))
+                            expect(error?.errorDescription).to(equal(expectedError.errorDescription))
+                            done()
+                        }
+                    }
+                }
+            }
+            
+            context("kid validation") {
+                let jwt = generateJWT()
+                let expectedError = IDTokenSignatureValidator.ValidationError.missingPublicKey(kid: "key123")
+                
+                it("should fail if the kid is not present") {
+                    stub(condition: isJWKSPath(domain)) { _ in jwksResponse(kid: nil) }
+                    
+                    waitUntil { done in
+                        signatureValidator.validate(jwt) { error in
+                            expect(error).to(matchError(expectedError))
+                            expect(error?.errorDescription).to(equal(expectedError.errorDescription))
+                            done()
+                        }
+                    }
+                }
+                
+                it("should fail if the kid does not match") {
```

Comment:
not match what?

Reply:
Fixed in c99945de2ec0aead4b9c5c9993162de5e1f08dfa.